src/ape_compile/_cli.py

```diff
@@ -26,33 +41,55 @@ def cli(contracts, compile_all, display_size):
     a project is loaded. You do not have to manually trigger a recompile.
     """
 
-    # TODO
-    # * Look for local manifest
-    # * If not, look for local config
-    # * If no config assume everything is default
-    # * `contracts` allows selecting specific contracts to compile from manifest
-    # Config should override defaults. If no config value, use default.
-    # default config is the source of truth, the manifest is an artifact of the
-    # config, both pre-and post procesing
-
-    if len(project.sources) == 0:
-        notify("ERROR", "Manifest contains no sources to compile")
+    # Expand source tree based on selection
+    if not filepaths:
+        if not (Path.cwd() / "contracts").exists():
+            notify("ERROR", "No `contracts/` directory detected")
+            return
+
+        # If no paths are specified, use all local project sources
+        contract_filepaths = project.sources
+
+    else:
+        # Expand any folder paths
+        expanded_filepaths = flatten([d.rglob("*.*") if d.is_dir() else [d] for d in filepaths])
+        # Filter by what's in our project's source tree
+        # NOTE: Make the paths absolute like `project.sources`
+        contract_filepaths = [
+            c.resolve() for c in expanded_filepaths if c.resolve() in project.sources
+        ]
+
+    if not contract_filepaths:
+        if filepaths:
+            selected_paths = "', '".join(str(p.resolve()) for p in filepaths)
+
+        else:
+            selected_paths = str(Path.cwd() / "contracts")
+
+        notify("WARNING", f"No project files detected in '{selected_paths}'")
         return
 
-    # NOTE: This compiles all types in `contracts/`
-    contract_types = list(project.contracts.values())
+    # TODO: only compile selected contracts
+    contract_types = project.load_contracts(use_cache)
```

Comment:
@lost-theory still need a better API for only compiling the selected files, but this is good enough for now I think